test/framework/src/main/java/org/elasticsearch/transport/nio/channel/AbstractNioChannel.java

```diff
@@ -35,14 +35,12 @@
  * A channel is open once it is constructed. The channel remains open and {@link #isOpen()} will return
  * true until the channel is explicitly closed.
  * <p>
- * A channel lifecycle has four stages:
+ * A channel lifecycle has three stages:
  * <ol>
- * <li>UNREGISTERED - When a channel is created and prior to it being registered with a selector.
- * <li>REGISTERED - When a channel has been registered with a selector. This is the state of a channel that
- * can perform normal operations.
+ * <li>OPEN - When a channel has been created. This is the state of a channel that can perform normal operations.
  * <li>CLOSING - When a channel has been marked for closed, but is not yet closed. {@link #isOpen()} will
  * still return true. Normal operations should be rejected. The most common scenario for a channel to be
- * CLOSING is when channel that was REGISTERED has {@link #closeAsync()} called, but the selector thread
```

Comment:
do we really need this `CLOSING` state or can we get rid of it. what is it's purpose? maybe only for testing? If so why don't we go with a simple boolean / AtomicBoolean and a test-only pkg private boolean that is indicating if we are fully closed yet?